src/gic_v2.rs

```diff
@@ -444,3 +444,67 @@ impl GicCpuInterface {
         self.regs().PMR.set(0xff);
     }
 }
+
+// define the GicTrait by crate_interface
+#[crate_interface::def_interface]
+pub trait GicTrait {
+    fn set_enable(vector: usize, enable: bool);
+    fn get_enable(vector: usize) -> bool;
+    fn get_typer() -> u32;
+    fn get_iidr() -> u32;
+
+    fn set_state(int_id: usize, state: usize, current_cpu_id: usize);
+    fn get_state(int_id: usize) -> usize;
+
+    fn set_icfgr(int_id: usize, cfg: u8);
+
+    fn get_target_cpu(int_id: usize) -> usize;
+    fn set_target_cpu(int_id: usize, target: u8);
+
+    fn get_priority(int_id: usize) -> usize;
+    fn set_priority(int_id: usize, priority: u8);
+}
+
+/// Implementation of [`GicTrait`] by crate_interface::call_interface,
```

Comment:
This is extremely strange.

You should `impl GicTrait for GicInterface`

and use methods provied by `GicInterface` for the implementation of `GicTrait`

Why did you call `call_interface!(xxx)` here...

It should by called by vgic, maybe?

Reply:
> This is extremely strange.
> 
> ~You should `impl GicTrait for GicInterface`~
> 
> ~and use methods provied by `GicInterface` for the implementation of `GicTrait`~
> 
> Why did you call `call_interface!(xxx)` here...
> 
> It should by called by vgic, maybe?

Sorry for mis-understanding, `GicTrait` should implemented somewhere with the **GIC instance**, which is hold by arceos

Reply:
We added another layer of abstraction on top of the original Trait, which has two benefits:  
1. It avoids exposing all the contents of `gic_v2`, such as `gic_v2::*`.  
2. We can simply use `GicInterface::set_enable()` without needing to import unnecessary `crate_interface`, achieving minimal dependency.

Reply:
> We added another layer of abstraction on top of the original Trait, which has two benefits:
> 
> 1. It avoids exposing all the contents of `gic_v2`, such as `gic_v2::*`.
> 2. We can simply use `GicInterface::set_enable()` without needing to import unnecessary `crate_interface`, achieving minimal dependency.

I have no problem with that,

`arm_vgic` can import `GicInterface` provided by `arm_gicv2` directlty.